src/main/java/org/terasology/metalrenegades/ai/system/CitizenDeathSystem.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.metalrenegades.ai.system;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.EventPriority;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.AliveCharacterComponent;
+import org.terasology.logic.characters.CharacterComponent;
+import org.terasology.logic.characters.CharacterTeleportEvent;
+import org.terasology.logic.health.BeforeDestroyEvent;
+import org.terasology.logic.health.DestroyEvent;
+import org.terasology.logic.health.HealthComponent;
+import org.terasology.logic.health.event.RestoreFullHealthEvent;
+import org.terasology.logic.players.PlayerCharacterComponent;
+import org.terasology.metalrenegades.ai.event.CitizenDeathEvent;
+import org.terasology.metalrenegades.minimap.events.RemoveCharacterFromOverlayEvent;
+import org.terasology.registry.In;
+import org.terasology.world.time.WorldTimeEvent;
+
+/**
+ * System to handle Citizen Deaths within Metal Renegades
+ */
+
+@RegisterSystem(value = RegisterMode.AUTHORITY)
+public class CitizenDeathSystem extends BaseComponentSystem {
+
+    Logger logger = LoggerFactory.getLogger(CitizenDeathSystem.class);
+
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_HIGH)
+    public void onCitizenDeath(BeforeDestroyEvent event, EntityRef entityRef,
```

Comment:
Right now, this event fires any time that **any** entity is destroyed. It fires for citizens, but also fires for killed deer, cups in the water system, and for players on death. It would be best to restrict this specifically to citizens, probably by restricting it to a particular component in the @RecieveEvent statement.

Reply:
Elaborating on the comment above, the entire game crashes with the below message when a player dies. This results from onCitizenDeath firing when the player entity is destroyed.

```
02:22:47.427 [main] INFO  o.t.m.m.AddCharacterPositionOverlaySystem - EntityRef{id = 655, netId = 0, prefab = 'engine:player'} has been remo
ved from the minimap
02:22:47.427 [main] INFO  o.t.m.ai.system.CitizenDeathSystem - EntityRef{id = 655, netId = 0, prefab = 'engine:player'} has died
02:22:47.512 [main] ERROR o.terasology.engine.TerasologyEngine - Uncaught exception, attempting clean game shutdown
java.lang.NullPointerException: null
        at org.terasology.logic.players.LocalPlayerSystem.processInput(LocalPlayerSystem.java:168)
        at org.terasology.logic.players.LocalPlayerSystem.update(LocalPlayerSystem.java:153)
        at org.terasology.engine.modes.StateIngame.update(StateIngame.java:171)
        at org.terasology.engine.TerasologyEngine.tick(TerasologyEngine.java:464)
        at org.terasology.engine.TerasologyEngine.mainLoop(TerasologyEngine.java:427)
        at org.terasology.engine.TerasologyEngine.run(TerasologyEngine.java:403)
        at org.terasology.engine.Terasology.main(Terasology.java:163)

```